### PR TITLE
Service hijacking scenario

### PR DESCRIPTION
**Pull request by tsebastiani**
This scenario is meant to hijack a `Service` traffic towards a custom webservice able to simulate time based status and payload responses based on a test plan provided in the krkn configuration:
 ```
service_target_port: http-web-svc
service_name: nginx-service
service_namespace: default
image: quay.io/redhat-chaos/krkn-service-hijacking:v0.1.0
chaos_duration: 300
plan:
  - resource: "/list/index.php"
    steps:
      GET:
        - duration: 5
          status: 500
          mime_type: "application/json"
          payload: |
            {
              "status":"internal server error"
            }
        - duration: 5
          status: 201
          mime_type: "application/json"
          payload: |
            {
              "status":"resource created"
            }
      POST:
        - duration: 7
          status: 401
          mime_type: "application/json"
          payload: |
            {
               "status": "unauthorized"
            }
        - duration: 1
          status: 404
          mime_type: "text/plain"
          payload: "not found"

  - resource: "/patch"
    steps:
      PATCH:
        - duration: 3
          status: 201
          mime_type: "text/plain"
          payload: "resource patched"
        - duration: 2
          status: 400
          mime_type: "text/plain"
          payload: "bad request"
```

Starting from the custom service pod start the plan will be executed for each resource defined for each method respecting the duration of each step defined.

### CODE REVIEW

File: run_kraken.py [hunk 1]
```diff
@@ -348,6 +349,9 @@ def main(cfg):
                         elif scenario_type == "network_chaos":
                             logging.info("Running Network Chaos")
                             failed_post_scenarios, scenario_telemetries = network_chaos.run(scenarios_list, config, wait_duration, kubecli, telemetry_k8s)
+                        elif scenario_type == "service_hijacking":
+                            logging.info("Running Service Hijacking Chaos")
+                            service_hijacking_plugin.run(scenarios_list,kubecli)
```
Comment (chaitanyaenr) [inline run_kraken.py:354]: Hitting TypeError: run() missing 1 required positional argument: 'telemetry'. 

Has to be " service_hijacking_plugin.run(scenarios_list,kubecli, telemetry_k8s)". Thoughts?

Comment (tsebastiani) [reply run_kraken.py]: Sorry I was pushing while you were testing.

File: scenarios/kube/service_hijacking.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# This scenario allows the user to simulate custom HTTP replies for a kubernetes Service.
```
Comment (paigerube14) [inline scenarios/kube/service_hijacking.yaml:1]: Lets make a readme doc for this with most of these details here

Comment (chaitanyaenr) [reply scenarios/kube/service_hijacking.yaml:1]: Example: https://github.com/krkn-chaos/krkn/blob/main/docs/zone_outage.md

File: run_kraken.py (hunk 1 above)
Comment (chaitanyaenr) [reply run_kraken.py]: Np at all @tsebastiani! It's working as expected!

File: scenarios/kube/service_hijacking.yaml (hunk 2 above)
Comment (tsebastiani) [reply scenarios/kube/service_hijacking.yaml:1]: :+1:

**Review by chaitanyaenr — APPROVED** [review]
LGTM!

**Pull request merged**